jdb/assignment1/q2-linkedlist.py

```diff
@@ -0,0 +1,58 @@
+class Node(object):
+  def __init__(self, value=None):
+    self.value, self.next = value, None
+
+
+class LinkedList(object):
+  def __init__(self, original_list):
+    current = self.head = Node(original_list[0])
+    for value in original_list[1:]:
+      current.next = Node(value)
+      current = current.next
+
+  def __iter__(self):
```

Comment:
Thank you, it was useful for me to look at this iterator implementation. I didn't do it by myself. Also I found useful this material about iterators in python, which also may be interesting for other trainee if they also want to go a bit deeper into the matter:

https://www.programiz.com/python-programming/iterator

Reply:
the iterator is tricky: there are two conditions for a class to work correctly with for loops:
1. __iter__ must be defined
2. __iter__ must return an iterator object, that is a object that has a __next__ method and the __next__ method must raise a StopIteration exception to signal the end of the loop

In my example: the __iter__ method is using yield, so automatically it is an iterator.